endpoints/directory_list_generator.py

```diff
@@ -46,6 +47,10 @@ def hello(self, request):
   implemented by HelloService.
   """
 
+  def __init__(self, request=None):
```

Comment:
We're recreating the generator each time we generate a directory list, because the request might come with a different URL or protocol or something.